source/common/protobuf/utility.cc

```diff
@@ -12,10 +13,19 @@ MissingFieldException::MissingFieldException(const std::string& field_name,
           fmt::format("Field '{}' is missing in: {}", field_name, message.DebugString())) {}
 
 void MessageUtil::loadFromFile(const std::string& path, Protobuf::Message& message) {
-  const std::string json = Filesystem::fileReadToEnd(path);
-  const auto status = Protobuf::util::JsonStringToMessage(json, &message);
+  const std::string file_contents = Filesystem::fileReadToEnd(path);
+  // If the filename ends with .pb, do a best-effort attempt to parse it as a proto.
+  if (StringUtil::endsWith(path, ".pb")) {
+    // Attempt to parse the binary format.
+    bool status = message.ParseFromString(file_contents);
+    if (status) {
+      return;
```

Comment:
raising an exception resulted in a bunch of tests failing.  //test/common/config:filesystem_subscription_impl_test
//test/common/config:subscription_impl_test
I believe it's because there's a file eds.pb which is sometimes (usually?) json.  renamed to eds.json since it appears to be.